src/main/algorithms/search/algorithms/breadthSearch.ts

```diff
@@ -0,0 +1,111 @@
+import type { SearchAlgorithmImplementation, SearchSimulation, SearchSimulationStep } from '@/main/algorithms/search/algorithms/types'
+import type { Graph } from '@/main/algorithms/search/graph/graph'
+import type { Coords, EdgeValue, GraphFormGrid, VertexValue } from '@/main/algorithms/search/graphForm/types'
+import type { Vertex } from '@/main/algorithms/search/graph/vertex'
+import { ProtocolBuilder } from '@/main/simulation/protocolBuilder'
+import { GraphFormItem } from '@/main/algorithms/search/graphForm/graphFormItem'
+import type { Edge } from '@/main/algorithms/search/graph/edge'
+
+export class BreadthSearch implements SearchAlgorithmImplementation {
+    run(graph: Graph<VertexValue, EdgeValue>, grid: GraphFormGrid, start: Vertex<VertexValue>, end: Vertex<VertexValue>): SearchSimulation {
```

Comment:
use the reaching of the end Vertex as a possible end condition

Reply:
What do you mean with "possible"? By now the breadth search stops always if the end vertex is reached.

Reply:
I mean that it does not get stuck, when the end isn't reachable.